BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -0,0 +1,1041 @@
+
+                                Vector3 toTarget;
+                                fc.throttle = 0;
+
+                                while (UnderTimeLimit() && targetVessel != null && !complete)
+                                {
+                                    toTarget = FromTo(vessel, targetVessel);
+                                    complete = toTarget.magnitude > minRange;
+                                    fc.attitude = toTarget.normalized;
+
+                                    yield return new WaitForFixedUpdate();
+                                }
+                            }
+                            else
+                            {
+                                currentStatus = "Maneuvering (Drift)";
+                                fc.throttle = 0;
+                                fc.attitude = Vector3.zero;
+                            }
+                        }
+                        else
+                        {
+                            currentStatus = "Stranded";
+                            fc.throttle = 0;
+                            fc.attitude = Vector3.zero;
+                        }
+
+                        yield return new WaitForSeconds(updateInterval);
+                    }
+                }
+            }
+            else
+            {
+                // Idle
+
+                if (hasWeapons)
+                    currentStatus = "Idle";
+                else
+                    currentStatus = "Idle (Unarmed)";
+
+                fc.throttle = 0;
+                fc.attitude = Vector3.zero;
+
+                yield return new WaitForSeconds(updateInterval);
+            }
+        }
+
+        #endregion Actual AI Pilot
+
+        #region Utility Functions
+
+        private bool CheckWithdraw()
+        {
+            var nearest = BDATargetManager.GetClosestTarget(weaponManager);
+            if (nearest == null) return false;
+
+            return Mathf.Abs(RelVel(vessel, nearest.Vessel).magnitude) < 200;
+        }
+
+        private bool CheckOrbitUnsafe()
+        {
+            Orbit o = vessel.orbit;
+            CelestialBody body = o.referenceBody;
+            double maxTerrainHeight = 200;
+            if (body.pqsController)
+            {
+                PQS pqs = body.pqsController;
+                maxTerrainHeight = pqs.radiusMax - pqs.radius;
+            }
+            minSafeAltitude = Math.Max(maxTerrainHeight, body.atmosphereDepth);
+
+            return (o.PeA < minSafeAltitude && o.timeToPe < o.timeToAp) || o.ApA < minSafeAltitude;
+        }
+
+        private bool UnderTimeLimit(float timeLimit = 0)
+        {
+            if (timeLimit == 0)
+                timeLimit = updateInterval;
+
+            return Time.time - lastUpdate < timeLimit;
+        }
+
+        private bool NearIntercept(Vector3 relVel, float minRange)
+        {
+            float timeToKillVelocity = relVel.magnitude / maxAcceleration;
```

Comment:
This should handle the case of `maxAcceleration = 0` unless it's always checked that it's non-zero before calling this function.